File: HLC01_03Java/src/main/java/org/example/gestionEstudiantes04/Teacher.java
package org.example.gestionEstudiantes04;

import java.util.Arrays;
import java.util.List;
import java.util.Scanner;


public class Teacher {
    private String nombre;
    private List<Student> students;
    private List<Course> courses;

    public Teacher(String nombre) {
        this.nombre = nombre != null ? nombre : "Profesor";
        this.students = Arrays.asList(
                new Student("Pepe"),
                new Student("Juan"),
                new Student("Ana")
        );
        this.courses = Arrays.asList(
                new Course("Matematicas"),
                new Course("Lengua")
        );
    }


    private String askForStudent() {
        System.out.println("Introduce el nombre del estudiante: ");
        Scanner scanner = new Scanner(System.in);
        return scanner.nextLine();
    }


    private String askForCourse() {
        System.out.println("Introduce el nombre del curso: ");
        Scanner scanner = new Scanner(System.in);
        return scanner.nextLine();
    }


    public void addStudent() {
        String studentName = askForStudent();
        Student student = students.stream()
                .filter(s -> s.getName().equals(studentName))
                .findFirst()
                .orElse(null);

        String courseName = askForCourse();
        Course course = courses.stream()
                .filter(c -> c.getName().equals(courseName))
                .findFirst()
                .orElse(null);

        if (course != null && student != null) {
            course.addStudent(student);
        } else {
            System.out.println("No se pudo añadir el estudiante al curso");
        }
    }


    public void showStudentGrades() {
        String studentName = askForStudent();
        Student student = students.stream()
                .filter(s -> s.getName().equals(studentName))
                .findFirst()
                .orElse(null);

        if (student != null) {
            Double grades = student.calcularPromedio();
            System.out.println("La nota media de " + student.getName() + " es de " + grades);
        } else {
            System.out.println("No se pudo calcular la nota");
        }
    }


    public void showStudents() {
        System.out.println("Estudiantes:");
        for (Student student : students) {
            System.out.println(student.getName());
        }
    }


    public void showCourses() {
        System.out.println("Cursos:");
        for (Course course : courses) {
            System.out.println(course.getName());
        }
    }


    public static void main(String[] args) {
        Teacher teacher = new Teacher("Profesor");
        teacher.showStudents();
        teacher.showCourses();
        teacher.addStudent();
        teacher.showStudentGrades();
    }

}

File: HLC01_03Kotlin/src/main/kotlin/gestionEstudiantes04/Teacher.kt
package org.example.gestionEstudiantes04


class Teacher(val nombre: String = "Profesor") {
    private val students = listOf(Student("Pepe"), Student("Juan"), Student("Ana"))
    private val courses = listOf(Course("Matematicas"), Course("Lengua"))

    private fun askForStudent(): String {
        println("introduce el nombre del estudiante: ")
        val student = readln()
        return student
    }


    private fun askForCourse(): String {
        println("introduce el nombre del curso: ")
        val course = readln()
        return course
    }


    fun addStudent() {
        val studentName = askForStudent()
        val student = students.find { it.name == studentName }

        val courseName = askForCourse()
        val course = courses.find { it.name == courseName }

        if (course != null && student != null) {
            course.addStudent(student)
        } else {
            println("No se pudo añadir el estudiante al curso")
        }
    }


    fun showStudentGrades() {
        val studentName = askForStudent()
        val student = students.find { it.name == studentName }
        if (student != null) {
            val grades = student.calcularPromedio()
            println("la nota media de ${student.name} es de $grades")
        } else {
            println("No se pudo calcular la nota")
        }
    }


    fun showStudents() {
        println("Estudiantes:")
        for (student in students) {
            println(student.name)
        }
    }


    fun showCourses() {
        println("Cursos:")
        for (course in courses) {
            println(course.name)
        }
    }

}
